docs/tutorials/custom_puzzle_secure.md

```diff
@@ -0,0 +1,82 @@
+---
+id: custom_puzzle_secure
+title: How to secure coin with a custom puzzle
+sidebar_label: Secure coin with a custom puzzle
+---
+
+This tutorial teaches you how to secure a coin with a custom-made puzzle correctly.
+
+## Create a puzzle
+Create whatever puzzle you want for your coin. I'll use a password-locked coin as an example for this tutorial.
+
+```chialisp
+(mod (password new_puzhash amount)
+    (defconstant CREATE_COIN 51)
+ 
+    (if (= (sha256 password) (q . 0x2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824))
+        (list (list CREATE_COIN new_puzhash amount))
+        (x)
+    )
+)
+```
+`0x2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824` is sha256 of the word `hello`. 
+
+**Example valid solution:** `(hello 0x5f5767744f91c1c326d927a63d9b34fa7035c10e3eb838c44e3afe127c1b7675 2)`.
+
+If password is correct, the puzzle will create a new coin with 2 mojos and secure it using a new puzzle hash `0x5f5767744f91c1c326d927a63d9b34fa7035c10e3eb838c44e3afe127c1b7675`.
+
+**Any remaining change goes to a farmer as a fee**.
+
+## Get compiled version of the custom puzzle
+There are multiple ways how you can get compiled version of your custom puzzle.
+
+### Using clvm_tools (official)
+You can compile your custom puzzle from the terminal using the clvm_tools repository (included in the official Chia repository). All you need to do is to use the `run` command with your puzzle provided as an argument.
+
+```bash
+$ run '(mod (password new_puzhash amount)
+    (defconstant CREATE_COIN 51)
+ 
+    (if (= (sha256 password) (q . 0x2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824))
+        (list (list CREATE_COIN new_puzhash amount))
+        (x)
+    )
+)'
+```
+results in `(a (q 2 (i (= (sha256 5) (q . 0x2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824)) (q 4 (c 2 (c 11 (c 23 ()))) ()) (q 8)) 1) (c (q . 51) 1))`.
+
+### Using Chialisp web tool (unofficial)
+You can obtain a compiled version of the puzzle using [Chialisp web tool](https://clisp.surrealdev.com/). Paste your custom puzzle into the text area and hit **Compile**. The compiled version will appear in the **Run Output** section.
+
+## Get puzzle hash from a puzzle
+You can use Chialisp to get the hash of the Chialisp puzzle. The puzzle to get puzzle hash is following:
+
+```chialisp
+(mod (puzzle)
+    (defconstant TREE 1)
+    
+    (defun sha256tree1 (TREE)
+       (if (l TREE)
+           (sha256 2 (sha256tree1 (f TREE)) (sha256tree1 (r TREE)))
+           (sha256 1 TREE)
+       )
+    )
+    
+    (sha256tree1 puzzle)
+)
+```
```

Comment:
Instead of this, maybe use `opc` with the `-H` option which can get you the tree hash of the script you pass to it.

Reply:
I've added both because this method also gives more context (which IMO helps with understanding). If you don't like it, I can also remove it, of course.

Reply:
I've also added this way as an option to get a serialized versions of the puzzle and solution.